lib/selections.js

```diff
@@ -53,28 +65,26 @@ export class Selections {
    * @param {SelectionItem & NotificationItem} newItem
    */
   insert (newItem) {
-    const { from, to, notify, isStreamSelection, ...restItem } = newItem
-    if (from > to) {
+    if (newItem.from > newItem.to) {
       throw new Error('Invalid interval')
     }
-    if (!isStreamSelection) {
-      this.remove({ from, to, isStreamSelection })
-    }
-    this._items.push({ from, to, isStreamSelection, ...restItem })
-    if (notify) this._notifications.push({ from, to, notify })
-  }
-
-  /** @param {MinimalSelectionItem} range */
-  callNotifyInRange (range) {
-    for (const notification of this._notifications) {
-      if (isLowerIntersecting(range, notification) ||
-            isUpperIntersecting(range, notification) ||
-            isInsideExisting(range, notification) ||
-            isCoveringExisting(range, notification)
-      ) {
-        notification.notify()
+    newItem.notify = newItem.notify ?? noop
+    if (!newItem.isStreamSelection) {
+      const { notify: oldNotify } = newItem
+      // Merge notifications of intersecting items into the new item's notify function
+      const intersectingNotifyFunctions = new Set(
```

Comment:
is there any point in de-duplicating them? it's unlikely it will ever happen and it forces you to iterate over the items 3 times, instead of once

Reply:
It was only there to merge all the noops, since they would all point to the same reference. But since noops are removed, I replaced this with a single `for..of` loop.